src/python/pants/backend/python/target_types.py

```diff
@@ -858,6 +858,23 @@ class PythonTestsExtraEnvVarsField(TestExtraEnvVarsField):
     pass
 
 
+class PythonTestsXdistConcurrencyField(IntField):
+    alias = "xdist_concurrency"
+    help = softwrap(
+        """
+        Number of CPUs that should be used to run each test file belonging to this target.
+
+        Tests are spread across multiple CPUs using `pytest-xdist`
+        (https://pytest-xdist.readthedocs.io/en/latest/index.html).
+        Use of `pytest-xdist` must be enabled using the `[pytest].xdist_enabled` option for
+        this field to have an effect.
+
+        If `pytest-xdist` is enabled and this field is unset, Pants will attempt to derive
+        the concurrency for test sources by counting the number of tests in each file.
```

Comment:
I know we'd eb repeating `pytest-xdist` docs, but I think we should also call out how shared fixtures are affected when enabling this option.

About once a day, we have people (internally and externally) asking about sharing a fixture across multiple test files, so this might increase that confusion.

Reply:
How is it impacted? We still can't share fixtures across files yet :/

Reply:
In this case session or class fixtures might not be shared across test functions (you'd have to read the xdist docs to see if I'm lying)

Reply:
Yeah OK my assumption is correct: https://pypi.org/project/pytest-xdist/#making-session-scoped-fixtures-execute-only-once

So even in a single test file, you wont' share a `session` fixture.

Reply:
I can add a note/link to the `help` text of the subsystem option

Reply:
I'll also add a note to the field `help` that you can set `xdist_concurrency=0` to explicitly disable it for a `python_test` (and a test showing this actually works the way I expect 😅)